### ISSUE TITLE
Autopilot consumes 100% of CPU

### ISSUE THREAD

**Issue by JaviLib** [lightningnetwork/lnd#5684]
### Background

After enabling autopilot (and wtcclient, although I don't know if it is related) with the following options:

```
autopilot.active=1
autopilot.maxchannels=1000
autopilot.allocation=1
autopilot.minchansize=900000
autopilot.maxchansize=1010000
;Autopilot.OpenChanFeeRate=11
autopilot.conftarget=1000
autopilot.private=false
;autopilot.heuristic=preferential:1
autopilot.heuristic=top_centrality:1.0
wtclient.active=true

```
My CPU goes 100% most of the time.

### Your environment

* version of `lnd` 0.13.1
* which operating system (`uname -a` on *Nix) Linux debian 5.10.0-8-amd64 
![imagen](https://user-images.githubusercontent.com/6039997/131760313-e5e4d462-65f8-4724-9b61-36f942f3b653.png)

**Comment by Roasbeef** [comment]
Are you able to extract a CPU profile? https://github.com/lightningnetwork/lnd/blob/a329c8061266418bccd797aa5d7d277c736ee930/docs/debugging_lnd.md#capturing-pprof-data-with-lnd

If you modify `allocation` to say, `0.90` does the issue disappear? 

What do you generally see in the logs? Particularly if you make the `ATPL` sub-system to the trace log level?

**Comment by JavierRSobrino** [comment]
Yes, after decreasing `allocation` to 0.90 the problem disappears.
Do you think it is still a bug? It should stop looking for channels if there is not enough onchain funds. It is absurd to continue to waste CPU if there is no more funds to open channels. In fact I think searching for channels for autopilot should be a low priority task, no need to waste so much CPU even if there are enough funds.

**Comment by bhandras** [comment]
Did you have available funds in your wallet above `autopilot.minchansize`?

**Comment by JavierRSobrino** [comment]
> Did you have available funds in your wallet above `autopilot.minchansize`?

I did in the beginning but after many open channels by autopilot it didn’t.
